### PR TITLE
Kill mutations in env

### PR DESCRIPTION
**Pull request by mbj**

### CODE REVIEW

**Comment by mbj** [comment]
@dkubb Ready for review after #490. Sorry for PR dependency, but its better to include the global fix here so a simple rebase after merge of dependant PR is sufficient.

**Comment by mbj** [comment]
@dkubb Ready for review.

File: spec/unit/mutant/env/boostrap_spec.rb [hunk 1]
```diff
@@ -155,14 +152,11 @@ def name
...
-        expected_warnings = [
+      before do
+        expected_warning =
           "Class#name from: #{invalid_class.inspect} returned Object. #{Mutant::Env::SEMANTICS_MESSAGE}"
-        ]
 
-        expect { subject }
-          .to change { config.reporter.warn_calls }
-          .from([]).to(expected_warnings)
+        expect(config.reporter).to receive(:warn).with(expected_warning).and_return(config.reporter)
```
Comment (dkubb) [inline spec/unit/mutant/env/boostrap_spec.rb]: The formatting here is inconsistent with other changes in this file above.

File: spec/unit/mutant/env/boostrap_spec.rb [hunk 2]
```diff
@@ -92,16 +91,14 @@ def name
         end
       end
 
-      it 'warns via reporter' do
-        expected_warnings = [
+      before do
+        expected_warning =
```
Comment (dkubb) [inline spec/unit/mutant/env/boostrap_spec.rb]: I would probably move `expected_warning` into a let statement.

File: spec/unit/mutant/env/boostrap_spec.rb [hunk 3]
```diff
@@ -155,14 +152,11 @@ def name
...
```
Comment (dkubb) [inline spec/unit/mutant/env/boostrap_spec.rb]: Instance of https://github.com/mbj/mutant/pull/491/files#r44890780

File: spec/unit/mutant/env/boostrap_spec.rb (hunk 1 above)
Comment (dkubb) [inline spec/unit/mutant/env/boostrap_spec.rb]: Given there are 3 placed in this spec that do the same thing, what about creating a single method that creates the expectation given `expected_warnings` and use it here and elsewhere?

File: spec/unit/mutant/env_spec.rb [hunk 4]
```diff
@@ -45,18 +45,38 @@
...
+          .and_return(mutation_subject)
+
+        expect(context).to receive(:root)
+          .with(s(:nil))
+          .and_return(wrapped_node)
+
+        expect(Mutant::Loader::Eval).to receive(:call)
+          .ordered
+          .with(wrapped_node, mutation_subject)
+          .and_return(nil)
```
Comment (dkubb) [inline spec/unit/mutant/env_spec.rb]: Is the return value used? If not, why not return a `double`? If it gets used it'll explode. I usually prefer to do this when defining expectations for methods called in "void context".

File: spec/unit/mutant/env/boostrap_spec.rb (hunk 1 above)
Comment (mbj) [inline spec/unit/mutant/env/boostrap_spec.rb]: Good catch. Fixed both.

File: spec/unit/mutant/env_spec.rb (hunk 4 above)
Comment (mbj) [inline spec/unit/mutant/env_spec.rb]: This is just a leftover. I replaced it with returning the receiver to mirror what the real `Mutant::Loader::Eval` does.

**Comment by mbj** [comment]
@dkubb Ready for the next pass.

**Comment by mbj** [comment]
@dkubb Assuming all green this is ready for the next pass.

File: spec/unit/mutant/env/boostrap_spec.rb [hunk 5]
```diff
@@ -41,12 +41,19 @@
...
+      .and_return(config.reporter)
+  end
+
   before do
     expect(integration_class).to receive(:new)
       .with(config)
       .and_return(integration)
 
-    expect(integration).to receive(:setup).and_return(integration)
+    expect(integration).to receive(:setup)
```
Comment (dkubb) [inline spec/unit/mutant/env/boostrap_spec.rb]: You can use `receive_messages` here.

File: spec/unit/mutant/env/boostrap_spec.rb [hunk 6]
```diff
@@ -146,6 +143,10 @@ def self.name
         end
       end
 
+      let(:expected_warning) do
+        "Class#name from: #{invalid_class.inspect} returned Object. #{Mutant::Env::SEMANTICS_MESSAGE}"
```
Comment (dkubb) [inline spec/unit/mutant/env/boostrap_spec.rb]: You wrapped the `expected_warning` above, but not here. I'd recommend wrapping.

File: spec/unit/mutant/env_spec.rb [hunk 7]
```diff
@@ -64,10 +84,18 @@
...
+          .and_raise(Mutant::Isolation::Error, 'test-error')
       end
 
-      let(:test_result) { Mutant::Result::Test.new(tests: tests, output: 'test-error', passed: false, runtime: 0.0) }
+      let(:test_result) do
+        Mutant::Result::Test.new(
+          tests:  tests,
+          output: 'test-error',
+          passed: false,
+          runtime: 0.0
```
Comment (dkubb) [inline spec/unit/mutant/env_spec.rb]: The hash keys should be aligned.

**Comment by mbj** [comment]
@dkubb Flags addressed, ready for the next pass.

File: spec/unit/mutant/env/boostrap_spec.rb [hunk 8]
```diff
@@ -75,6 +76,11 @@
     context 'when Module#name calls result in exceptions' do
       let(:object_space_modules) { [invalid_class] }
 
+      let(:expected_warning) do
+        "Class#name from: #{invalid_class} raised an error: " \
```
Comment (dkubb) [inline spec/unit/mutant/env/boostrap_spec.rb]: Below you use `invalid_class.inspect` but not here. Any reason for the difference?

Comment (mbj) [inline spec/unit/mutant/env/boostrap_spec.rb]: Just unaccounted variance. Reducing the other case to the minimum.

**Comment by mbj** [comment]
@dkubb Ready for the next pass.

**Pull request merged**